### PR TITLE
New function io_uring_sq_space_left

### PR DESCRIPTION
**Pull request by rouzier**
New function to calculate how much space you have left in the submission queue.

### CODE REVIEW

File: src/include/liburing.h [hunk 1]
```diff
@@ -237,6 +237,11 @@ static inline void io_uring_prep_timeout(struct io_uring_sqe *sqe,
 	io_uring_prep_rw(IORING_OP_TIMEOUT, sqe, 0, ts, 1, count);
 }
 
+static inline int io_uring_sq_space_left(struct io_uring *ring)
```
Comment (axboe) [inline src/include/liburing.h]: This should return an unsigned, not an integer.

File: test/sq-space_left.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	ret = io_uring_queue_init(8, &ring, 0);
+	if (ret) {
+		printf("ring setup failed\n");
+		return 1;
+
+	}
+
+	if ( (s = io_uring_sq_space_left(&ring)) != 8) {
+		printf("Got %d SQEs left\n", s);
```
Comment (axboe) [inline test/sq-space_left.c]: Please include expected result as well, ala:

Got %d SQEs left, expected %d

File: test/sq-space_left.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	int ret, i = 0, s;
+
+	ret = io_uring_queue_init(8, &ring, 0);
+	if (ret) {
+		printf("ring setup failed\n");
+		return 1;
+
+	}
+
+	if ( (s = io_uring_sq_space_left(&ring)) != 8) {
```
Comment (axboe) [inline test/sq-space_left.c]: No space between the parens here.

File: test/sq-space_left.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		printf("Got %d SQEs left\n", s);
+		goto err;
+	}
+
+	i = 0;
+	while ((sqe = io_uring_get_sqe(&ring)) != NULL) {
+		i++;
+		if ( (s = io_uring_sq_space_left(&ring)) != 8 - i) {
```
Comment (axboe) [inline test/sq-space_left.c]: Space again

File: test/sq-space_left.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			printf("Got %d SQEs left\n", s);
+			goto err;
+		}
+	}
+
+	if (i != 8) {
```
Comment (axboe) [inline test/sq-space_left.c]: Ditto on including expected result

**Comment by rouzier** [comment]
Done.

**Comment by axboe** [comment]
Pulled, thanks.

**Pull request merged**